src/cargo/ops/cargo_new.rs

```diff
@@ -140,14 +216,66 @@ fn mk(config: &Config, path: &Path, name: &str,
         (None, None, name, None) => name,
     };
 
-    try!(paths::write(&path.join("Cargo.toml"), format!(
+    let license: Option<Vec<License>> = match (&opts.license, cfg.license) {
+        (&None, None) => None,
+        (&Some(ref lic), _) => Some(lic.clone()),
+        (&_, Some(lic)) => Some(lic.clone()),
+    };
+
+    let license_file: Option<String> = match (&opts.license_file, cfg.license_file) {
+        (&None, None) => None,
+        (&Some(ref lic_file), _) => Some(lic_file.to_string()),
+        (&_, Some(lic_file)) => Some(lic_file.to_string()),
+    };
+
+    let license_options: String = match (license.clone(), license_file) {
+        (Some(license), Some(license_file)) => {
+            let license = join_licenses(license.iter()
+                                               .map(|l| format!("{}", l))
+                                               .collect::<Vec<_>>());
+            format!(r#"license = "{}"
+license-file = "{}"
+"#, license, license_file)
+        },
+        (Some(license), None) => {
+            let license = join_licenses(license.iter()
+                                               .map(|l| format!("{}", l))
+                                               .collect::<Vec<_>>());
+            format!(r#"license = "{}"
+"#, license)
+        },
+        (None, Some(license_file)) => {
+            format!(r#"license-file = "{}"
+"#, license_file)
+        },
+        (None, None) => {
+            "".to_owned()
```

Comment:
This may still want to be a newline to ensure that the generated Cargo.toml without these options is still the same